pkg/generic/thrift/parse.go

```diff
@@ -219,6 +219,20 @@ func addFunction(fn *parser.Function, tree *parser.Thrift, sDsc *descriptor.Serv
 		Response:       resp,
 		HasRequestBase: hasRequestBase,
 	}
+	err = handleRouter(fn, sDsc, fnDsc)
+	if err != nil {
+		return err
+	}
+	sDsc.Functions[fn.Name] = fnDsc
+	return nil
+}
+
+func handleRouter(fn *parser.Function, sDsc *descriptor.ServiceDescriptor, fnDsc *descriptor.FunctionDescriptor) (err error) {
```

Comment:
Don't need to and a new function.

Reply:
I don't know how to return err right after there's a panic without adding a new function...

Reply:
I don't know how to return err right after there's a panic without adding a new function...

Reply:
Add the definition of error, and then set err in the defer function

Reply:
@jayantxie Is it possible to **return** the error inside the defer function?